ragstack-e2e-tests/e2e_tests/conftest.py

```diff
@@ -42,15 +53,108 @@ class AstraRef:
     env: str
 
 
-def get_astra_ref() -> AstraRef:
-    env = os.environ.get("ASTRA_DB_ENV", "prod").lower()
-    return AstraRef(
-        token=get_required_env("ASTRA_DB_TOKEN"),
-        api_endpoint=get_required_env("ASTRA_DB_ENDPOINT"),
-        collection=get_required_env("ASTRA_TABLE_NAME"),
-        id=get_required_env("ASTRA_DB_ID"),
-        env=env,
-    )
+class VectorDatabaseHandler:
+    def __init__(self):
+        self.mode = os.environ.get("VECTOR_DATABASE_MODE", "astradb")
+        if self.mode not in ["astradb", "dse"]:
+            raise ValueError(f"Invalid VECTOR_DATABASE_MODE: {self.mode}")
+        self.test_table_name = None
+        self.cassandra_container = None
+
+    def is_astradb(self) -> bool:
```

Comment:
Can we have subclasses instead of checking `is_astradb ` everywhere ?
Something like `AstraDBVectorDatabaseHandler`, `DseVectorDatabaseHandler`

Reply:
With a `VectorDatabaseHandler.from(mode)` class method.